axelrod/result_set.py

```diff
@@ -609,6 +610,17 @@ def __eq__(self, other):
             other : axelrod.ResultSet
                 Another results set against which to check equality
         """
+        def list_equal(v1: List[float], v2: List[float]) -> bool:
```

Comment:
Can we say more specifically that nans are considered equal, and rename the function to something like `list_equal_with_nans` or something like that?